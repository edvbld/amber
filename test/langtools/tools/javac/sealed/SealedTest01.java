/*
 * @test /nodynamiccopyright/
 * @summary smoke test for sealed classes
 * @compile/fail/ref=SealedTest01.out -XDrawDiagnostics SealedTest01.java
 */

class SealedTest01 {
    static sealed class SC permits C_SC { }
    static sealed abstract class SAC permits C_SAC { }
    static sealed interface SI permits C_SI, I_SI { }

    static class C_SC extends SC { }
    static class C_SAC extends SAC { }
    static class C_SI implements SI { }
    static interface I_SI extends SI { }
}

class SealedTest01_Other {
    static class C_SC extends SealedTest01.SC { }
    static class C_SAC extends SealedTest01.SAC { }
    static class C_SI implements SealedTest01.SI { }
    static interface I_SI extends SealedTest01.SI { }
}
