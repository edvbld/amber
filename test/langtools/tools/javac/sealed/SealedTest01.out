SealedTest01.java:19:12: compiler.err.cant.inherit.from.sealed: SealedTest01.SC
SealedTest01.java:20:12: compiler.err.cant.inherit.from.sealed: SealedTest01.SAC
SealedTest01.java:21:12: compiler.err.cant.inherit.from.sealed: SealedTest01.SI
SealedTest01.java:22:12: compiler.err.cant.inherit.from.sealed: SealedTest01.SI
4 errors
